src/browser/base/zen-components/ZenPinnedTabManager.mjs

```diff
@@ -388,6 +391,9 @@
       for (let i = 0; i < tabs.length; i++) {
         const tab = tabs[i];
         tab.removeAttribute("zen-essential");
+        if(ZenWorkspaces.workspaceEnabled && ZenWorkspaces.getActiveWorkspaceFromCache.uuid) {
+          tab.setAttribute("zen-workspace-id", ZenWorkspaces.getActiveWorkspaceFromCache.uuid);
```

Comment:
Essentials dont care about workspaces, no?

Reply:
They do not. 
I modified `addToEssentials` to remove the workspace id so that essentials wouldn't be tied to a workspace in any way. When you are removing an essential using `removeEssentials`, that tab doesn't have a workspace id so it has to have one set - in this case the id of the current workspace